### PR TITLE
cppcheck: fix some reports

### PR DESCRIPTION
**Pull request by serval2412**
[net/monalisa/src/TMonaLisaWriter.cxx:956] -> [net/monalisa/src/TMonaLisaWriter.cxx:957]: (error) Iterator 'iter' used after element has been erased.
[proof/proofd/src/XrdProofdProofServMgr.cxx:4648] -> [proof/proofd/src/XrdProofdProofServMgr.cxx:4646]: (error) Iterator 'iter' used after element has been erased.

[math/minuit2/src/Minuit2Minimizer.cxx:669] -> [math/minuit2/src/Minuit2Minimizer.cxx:669]: (style) Same expression on both sides of '||'.
[math/minuit2/src/Minuit2Minimizer.cxx:736] -> [math/minuit2/src/Minuit2Minimizer.cxx:736]: (style) Same expression on both sides of '||'.
[math/minuit2/src/Minuit2Minimizer.cxx:753] -> [math/minuit2/src/Minuit2Minimizer.cxx:753]: (style) Same expression on both sides of '||'.

[tree/tree/src/TBasket.cxx:852]: (style) Redundant condition: If 'flag >= 80', the comparison 'flag' is always true.
[misc/table/src/TFileIter.cxx:467] -> [misc/table/src/TFileIter.cxx:468]: (warning) Identical condition 'thisRunNumber<runNumber', second condition is always false

[math/mathcore/src/TMath.cxx:826]: (style) Array index 'ia' is used before limits check.
[math/mathcore/src/TMath.cxx:830]: (style) Array index 'ib' is used before limits check.
[math/mathcore/src/triangle.c:15434]: (style) Array index 'aspectindex' is used before limits check.
[roofit/roofit/src/RooIntegralMorph.cxx:375]: (style) Array index 'igapHigh' is used before limits check.

[core/winnt/src/TWinNTSystem.cxx:5035]: (style) Statements following return, break, continue, goto or throw will never be executed.
[tree/treeplayer/src/TSelectorDraw.cxx:380]: (style) Statements following return, break, continue, goto or throw will never be executed.

[net/auth/src/TAuthenticate.cxx:4205]: (error) Resource leak: fd
[roofit/roofitcore/src/BidirMMapPipe.cxx:1880]: (error) Memory leak: s
[tree/tree/src/TLeaf.cxx:222]: (error) Memory leak: countname
[tree/tree/src/TLeaf.cxx:226]: (error) Memory leak: countname
[tmva/tmva/src/MethodBase.cxx:2783]: (error) Memory leak: pdfS
[tmva/tmva/src/MethodBase.cxx:2783]: (error) Memory leak: pdfB

[roofit/roofitcore/src/RooAbsArg.cxx:280]: (style) Redundant checking of STL container element existence before removing it.

[core/dictgen/src/rootcling_impl.cxx:457]: (performance) Possible inefficient checking for 'fieldSelRules' emptiness.
[core/dictgen/src/rootcling_impl.cxx:4573]: (performance) Possible inefficient checking for 'filesIncludedByLinkdef' emptiness.
[gui/canvaspainter/v7/src/TCanvasPainter.cxx:322]: (performance) Possible inefficient checking for 'fWebConn' emptiness.
[gui/canvaspainter/v7/src/TCanvasPainter.cxx:374]: (performance) Possible inefficient checking for 'fCmds' emptiness.
[gui/canvaspainter/v7/src/TCanvasPainter.cxx:419]: (performance) Possible inefficient checking for 'fWebConn' emptiness.
[gui/canvaspainter/v7/src/TCanvasPainter.cxx:475]: (performance) Possible inefficient checking for 'fWebConn' emptiness.
[gui/canvaspainter/v7/src/TCanvasPainter.cxx:587]: (performance) Possible inefficient checking for 'fCmds' emptiness.
[gui/canvaspainter/v7/src/TCanvasPainter.cxx:787]: (performance) Possible inefficient checking for 'fCmds' emptiness.
[html/src/TDocParser.cxx:684]: (performance) Possible inefficient checking for 'currentType' emptiness.
[proof/proofd/src/XrdProofdManager.cxx:804]: (performance) Possible inefficient checking for 'uwrks' emptiness.
[proof/proofd/src/XrdProofdSandbox.cxx:467]: (performance) Possible inefficient checking for 'staglst' emptiness.
[proof/proofd/src/XrdProofdSandbox.cxx:571]: (performance) Possible inefficient checking for 'actln' emptiness.
[roofit/roofitcore/src/RooSimWSTool.cxx:574]: (performance) Possible inefficient checking for 'slist' emptiness.

all the reports Prefer prefix ++/-- operators for non-primitive types

### CODE REVIEW

**Comment by Axel-Naumann** [comment]
Thanks, @serval2412 !
@phsft-bot build, please.

**Comment by serval2412** [comment]
Any update?

**Comment by amadio** [comment]
Sorry, and thanks for the reminder. I'll take care of it by Monday.

**Comment by amadio** [comment]
@phsft-bot build

**Comment by serval2412** [comment]
I can't see the failing logs because I must authenticate to CERN SSO so can't help here :-(

**Comment by amadio** [comment]
Don't worry, it's probably not your fault. I'll take care of it.

File: misc/table/src/TFileIter.cxx [hunk 1]
```diff
@@ -465,7 +465,7 @@ TKey *TFileIter::NextEventKey(UInt_t eventNumber, UInt_t runNumber, const char *
       if (runNumber != UInt_t(-1)) {
          UInt_t thisRunNumber = thisKey.RunNumber();
          if (thisRunNumber < runNumber) continue;
-         if (thisRunNumber < runNumber) { key = 0; break; }
+         if (thisRunNumber > runNumber) { key = 0; break; }
```
Comment (amadio) [inline misc/table/src/TFileIter.cxx]: @pcanal Could you please comment on this change? I am not sure about it.

File: proof/proofd/src/XrdProofdProofServMgr.cxx [hunk 2]
```diff
@@ -4642,10 +4642,10 @@ bool XrdProofdProofServMgr::Alive(XrdProofdProtocol* p)
       int rect = now - iter->second;
       if (rect < fReconnectTimeOut) {
          if (p == iter->first) alive = false;
+         ++iter;
       } else {
-         fDestroyTimes.erase(iter);
+         iter = fDestroyTimes.erase(iter);
       }
-      iter++;
```
Comment (amadio) [inline proof/proofd/src/XrdProofdProofServMgr.cxx]: @gganis Hi Gerri, could you please comment here? I'd like to merge these fixes, but I'm unsure about this particular change. Thanks!

File: misc/table/src/TFileIter.cxx (hunk 1 above)
Comment (pcanal) [reply misc/table/src/TFileIter.cxx]: LGTM

**Comment by amadio** [comment]
@phsft-bot build

**Comment by amadio** [comment]
Build passed, but waiting on windows, which is not there... Merging.

**Pull request merged**